### ISSUE TITLE
winapi/fileapi.h and hal/fileio.h file creation flags definitions overlap

### ISSUE THREAD

**Issue by mborgerson** [XboxDev/nxdk#256]
```
In file included from /home/mborgerson/Projects/xbox/nxdk/samples/mesh/../../lib/winapi/findfile.c:8:
/home/mborgerson/Projects/xbox/nxdk/samples/mesh/../../lib/winapi/fileapi.h:13:9: warning: 'CREATE_NEW' macro redefined
      [-Wmacro-redefined]
#define CREATE_NEW 1
        ^
/home/mborgerson/Projects/xbox/nxdk/samples/mesh/../../lib/hal/fileio.h:19:9: note: previous definition is here
#define CREATE_NEW                              0x00000001
        ^
In file included from /home/mborgerson/Projects/xbox/nxdk/samples/mesh/../../lib/winapi/findfile.c:8:
/home/mborgerson/Projects/xbox/nxdk/samples/mesh/../../lib/winapi/fileapi.h:14:9: warning: 'CREATE_ALWAYS' macro redefined
      [-Wmacro-redefined]
#define CREATE_ALWAYS 2
        ^
/home/mborgerson/Projects/xbox/nxdk/samples/mesh/../../lib/hal/fileio.h:20:9: note: previous definition is here
#define CREATE_ALWAYS                           0x00000002
        ^
In file included from /home/mborgerson/Projects/xbox/nxdk/samples/mesh/../../lib/winapi/findfile.c:8:
/home/mborgerson/Projects/xbox/nxdk/samples/mesh/../../lib/winapi/fileapi.h:15:9: warning: 'OPEN_EXISTING' macro redefined
      [-Wmacro-redefined]
#define OPEN_EXISTING 3
        ^
/home/mborgerson/Projects/xbox/nxdk/samples/mesh/../../lib/hal/fileio.h:21:9: note: previous definition is here
#define OPEN_EXISTING                           0x00000003
        ^
In file included from /home/mborgerson/Projects/xbox/nxdk/samples/mesh/../../lib/winapi/findfile.c:8:
/home/mborgerson/Projects/xbox/nxdk/samples/mesh/../../lib/winapi/fileapi.h:16:9: warning: 'OPEN_ALWAYS' macro redefined
      [-Wmacro-redefined]
#define OPEN_ALWAYS 4
        ^
/home/mborgerson/Projects/xbox/nxdk/samples/mesh/../../lib/hal/fileio.h:22:9: note: previous definition is here
#define OPEN_ALWAYS                             0x00000004
        ^
In file included from /home/mborgerson/Projects/xbox/nxdk/samples/mesh/../../lib/winapi/findfile.c:8:
/home/mborgerson/Projects/xbox/nxdk/samples/mesh/../../lib/winapi/fileapi.h:17:9: warning: 'TRUNCATE_EXISTING' macro redefined
      [-Wmacro-redefined]
#define TRUNCATE_EXISTING 5
        ^
/home/mborgerson/Projects/xbox/nxdk/samples/mesh/../../lib/hal/fileio.h:23:9: note: previous definition is here
#define TRUNCATE_EXISTING                       0x00000005
```

These should be defined in one place (probably just winapi/fileapi.h as we should drop the xapi file creation stuff for more standard interfaces).

**Comment by smashteevee** [comment]
What's the preferred fix here? Some options I think:

1. We could remove them from hal/fileio.h but other files seem to still rely solely on the define in hal/fileio (e.g,  _PDCLIB_open.c), which I guess we can point them away from. Not sure what new rabbithole we'd start to go down (new dependencies/conflicts to resolve), but that is a step closer to moving away from hal/fileio, if that is the long-term goal
2. We could add in some #ifndefs to wrap around the winapi/fileapi.h #defines. Not very graceful but easy 
3. We could update the #define values in winapi/fileapi.h to hex, which would shut the preprocessor  go down up (eg, it's warning because it sees 0x0000005 vs 5). Not introducing inconsistency but also not moving us further away from hal/fileio, which I assume is 

LMK if I'm making some wrong assumptions here.

**Comment by JayFoxRox** [comment]
> What's the preferred fix here? Some options I think: [...]

As suggested in https://github.com/XboxDev/nxdk/pull/150#issue-303958690 we'll probably remove hal/fileio.c (and the respective header). They had been deprecated since late August 2019.

The migration currently depends on https://github.com/XboxDev/nxdk-pdclib/pull/15.
I don't think we should touch anything until this is complete (which should be very soon).

**Comment by thrimbor** [comment]
Fixed by #385.
